test/unit/iteration_test.rb

```diff
@@ -3,7 +3,78 @@
 require "test_helper"
 
 class JobIterationTest < IterationUnitTest
+  class JobWithNoMethods < ActiveJob::Base
+    include JobIteration::Iteration
+  end
+
+  class JobWithRightMethods < ActiveJob::Base
+    include JobIteration::Iteration
+    def build_enumerator(_params, cursor:)
+      enumerator_builder.build_times_enumerator(2, cursor: cursor)
+    end
+
+    def each_iteration(*)
+    end
+  end
+
+  class JobWithRightMethodsButMissingCursorKeywordArgument < ActiveJob::Base
+    include JobIteration::Iteration
+    def build_enumerator(params, cursor)
+      enumerator_builder.active_record_on_records(
+        Product.where(id: params[:id]),
+        cursor: cursor,
+      )
+    end
+
+    def each_iteration(product, params)
+    end
+  end
+
+  class JobWithRightMethodsUsingSplatInTheArguments < ActiveJob::Base
+    include JobIteration::Iteration
+    def build_enumerator(*)
+    end
+
+    def each_iteration(*)
+    end
+  end
+
+  def test_jobs_that_define_build_enumerator_and_each_iteration_will_not_raise
+    push(JobWithRightMethods, 'walrus' => 'best')
+    work_one_job
+  end
+
+  def test_jobs_that_pass_splat_argument_to_build_enumerator_will_not_raise
+    push(JobWithRightMethodsUsingSplatInTheArguments, {})
+    work_one_job
+  end
+
+  def test_jobs_that_do_not_define_build_enumerator_or_each_iteration_will_not_raise
```

Comment:
`each_iteration_will_not_raise` but the test has an `assert_raises(ArgumentError)`